lib/archive/zip/entry.rb

```diff
@@ -481,8 +481,8 @@ def initialize(zip_path, raw_data = nil)
       self.expected_data_descriptor = nil
       self.compression_codec = Zip::Codec::Store.new
       self.encryption_codec = Zip::Codec::NullEncryption.new
-      self.password = nil
       @raw_data = raw_data
+      self.password = nil
```

Comment:
I also forgot why this change was needed... it looks unnecessary for this PR.

Reply:
Once I finally got in to look over the code, the need became clear.  The `password=` method uses `@rawdata`, so `@rawdata` needs to be set first.

Reply:
I figured it out when I finally got in to check the code.  It was necessary because the `password=` method uses `@rawdata`, so `@rawdata` needed to be initialized before setting the password.